sleap/gui/commands.py

```diff
@@ -1843,20 +1843,23 @@ class RemoveVideo(EditCommand):
     def do_action(context: CommandContext, params: dict):
         videos = context.labels.videos.copy()
         row_idxs = context.state["selected_batch_video"]
+        videos_to_be_removed = [videos[i] for i in row_idxs]
 
         # Remove selected videos in the project
         for idx in row_idxs:
+            context.labels.remove_video(videos[idx])
 
-            # check if video to be deleted is the current state video
-            if videos[idx] == context.state["video"]:
-                if len(context.labels.videos):
+        # check if video to be deleted is the current state video
+        if context.state["video"] in videos_to_be_removed:
+            if len(context.labels.videos):
+                try:
                     context.state["video"] = context.labels.videos[
                         list(set(range(len(videos))) - set(row_idxs))[-1]
                     ]
-                else:
+                except:
```

Comment:
Using a plain `try-except` without specifying the particular exception you're expecting to handle is a [common anti-pattern](https://realpython.com/the-most-diabolical-python-antipattern/).

Doing so is quick to write and handle complex control flow scenarios, but it has the side effect of obscuring errors that we didn't mean to catch. Instead I think what you want to do here is use a more specific error type, like:

```py
try:
    ...
except IndexError:
    # Comment on what the specific case you're catching is and what we're doing to handle it.
    ...
```

[Here's a list of common exceptions](https://docs.python.org/3/library/exceptions.html).

In this particular case, you could just do a test for whether the new video is available and just wrap this in an `if-else` block.

It also helps to document under what circumstances the error you're catching is raised, and what the new behavior is if that happens.

Like right now it looks like it defaults to switching to the last video, but I don't see that described anywhere.

Reply:
sure, I'll wrap it in an if-else condition